### PR TITLE
Renames in servermode per feedback

### PR DESCRIPTION
**Pull request by sheetalkamat**
Renames done as pert https://github.com/microsoft/TypeScript/pull/39735#pullrequestreview-457936086

Not sure if its late to do this but here is the PR

### CODE REVIEW

**Comment by DanielRosenwasser** [comment]
@typescript-bot cherry-pick this to release-4.0

**Comment by DanielRosenwasser** [comment]
@mjbvz should we change the TSServer flag as well? Or should we leave it alone and keep the rename as part of the API?

**Comment by amcasey** [comment]
Obviously, I'm fine with the names I suggested, but I'm not stuck to them.

File: src/tsserver/server.ts [hunk 1]
```diff
@@ -960,9 +960,9 @@ namespace ts.server {
             case "semantic":
                 return LanguageServiceMode.Semantic;
             case "approximatesemanticonly":
-                return LanguageServiceMode.ApproximateSemanticOnly;
+                return LanguageServiceMode.ApproximateSemantic;
             case "syntaxonly":
```
Comment (DanielRosenwasser) [inline src/tsserver/server.ts:964]: So this will be changed to `syntactic`

File: src/tsserver/server.ts [hunk 2]
```diff
@@ -960,9 +960,9 @@ namespace ts.server {
             case "semantic":
                 return LanguageServiceMode.Semantic;
             case "approximatesemanticonly":
```
Comment (DanielRosenwasser) [inline src/tsserver/server.ts:962]: And this will be changed to `partialsemantic`?

**Comment by sheetalkamat** [comment]
@typescript-bot cherry-pick this to release-4.0

**Comment by DanielRosenwasser** [comment]
@typescript-bot cherry-pick this to release-4.0 and LKG

**Comment by sheetalkamat** [comment]
@amcasey @DanielRosenwasser Ready to merge ?

**Pull request merged**